### PR TITLE
feat(LINGUIST-318): Default word list added on registration

### PR DESCRIPTION
**Pull request by KardelenCeren**
The new service method adds a predefined word list (from a yaml file). 

There is currently no image associated with the default list.

### CODE REVIEW

File: src/main/java/app/linguistai/bmvp/request/wordbank/QPredefinedWordList.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package app.linguistai.bmvp.request.wordbank;
+
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.NotNull;
+import lombok.Data;
```
Comment (can-aslan) [inline src/main/java/app/linguistai/bmvp/request/wordbank/QPredefinedWordList.java:5]: Removing unused imports can improve the readability and mainatainability of the code

File: src/main/java/app/linguistai/bmvp/service/wordbank/UnknownWordService.java [hunk 2]
```diff
@@ -343,6 +345,60 @@ public ROwnerUnknownWordList deleteList(UUID listId, String email) throws Except
         }
     }
 
+    @Transactional
+    public ROwnerUnknownWordList addPredefinedWordList(String wordListYamlFile, String email) throws Exception {
```
Comment (can-aslan) [inline src/main/java/app/linguistai/bmvp/service/wordbank/UnknownWordService.java:349]: Removing the `wordListYamlFile` parameter and using the defined constant insted in the function will be better in terms of the principle of least privilege, potentially avoiding some problems

File: src/main/java/app/linguistai/bmvp/utils/FileUtils.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package app.linguistai.bmvp.utils;
+
+import app.linguistai.bmvp.request.wordbank.QPredefinedWordList;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import java.io.InputStream;
+
+public class FileUtils {
+
+    public static QPredefinedWordList readPredefinedWordListFromYamlFile(String filePath) throws Exception {
```
Comment (can-aslan) [inline src/main/java/app/linguistai/bmvp/utils/FileUtils.java:10]: For loading quests from a yaml file, I used a different architecture (not in dev, in LINGUIST-...-Quest-... branch).

It looks like there should not be problems but have you tested that this works not only upon user microservice boot but also dynamically?

**Review by can-aslan — CHANGES_REQUESTED** [review]
LGTM, just left some minor comments 🚀, eline saglıkk

File: src/main/java/app/linguistai/bmvp/service/wordbank/UnknownWordService.java (hunk 2 above)
Comment (KardelenCeren) [reply src/main/java/app/linguistai/bmvp/service/wordbank/UnknownWordService.java:349]: Keeping it so we can add more predefined word lists in the future, for IELTS etc.

File: src/main/java/app/linguistai/bmvp/utils/FileUtils.java (hunk 3 above)
Comment (KardelenCeren) [reply src/main/java/app/linguistai/bmvp/utils/FileUtils.java:10]: The new default list is only added when registering for a new user, so yes.
Users created before this feature will not have a default word list

**Review by can-aslan — APPROVED** [review]
LGTM 🔥

**Pull request merged**